app/src/main/java/de/othr/plantico/ui/PlantViewModel.kt

```diff
@@ -20,6 +20,13 @@ class TestViewModel(private val repository: Repository) : ViewModel() {
     fun insertOwnedPlant(plant: OwnedPlant)= viewModelScope.launch {
         repository.insertOwnedPlant(plant)
     }
+
+    companion object {
+        //Search for plants that contain the query as a substring. Not case-sensitive!
+        fun searchForPlantsInList(plants: List<Plant>, query: String): List<Plant> {
+            return plants.filter { plant: Plant -> plant.plantName.contains(query, ignoreCase = true) }
+        }
+    }
```

Comment:
This does not make sense. A ViewModel contains the data, in this case the plantList, and passes it to the Activity. Here, you reverse this order, which is unnecessary as the ViewModel already has access to the plantList. Either use the ViewModel's data here, or move it to the Activity. In the Activity it shouldn't be static as well, as you should use the plantList property you defined there.